packages/components/src/Layout/Semantics/Aside.tsx

```diff
@@ -52,16 +53,28 @@ export interface AsideProps extends SemanticLayoutBase, SemanticBorderProps {
   width?: ResponsiveValue<AsideSizeRamp | string>
 }
 
-const AsideLayout: FC<AsideProps> = ({ collapse, ...props }) =>
-  collapse ? null : <aside {...props} />
+export const Aside: FC<AsideProps> = ({ collapse, children, ...props }) => {
+  const internalRef = useRef<HTMLDivElement>(null)
+  const hasOverflow = useOverflow(internalRef)
 
-export const Aside = styled(AsideLayout)
-  .withConfig({
+  return collapse ? null : (
+    <InnerAside hasOverflow={hasOverflow} ref={internalRef} {...props}>
```

Comment:
Create a single component that be used in all the components (see mention of `OverflowShadow` below:

```suggestion
    <OverflowShadow as="aside" hasOverflow={hasOverflow} ref={internalRef} {...props}>
```